ui/app/src/components/SiteToolsApp/UrlDrivenSiteTools.tsx

```diff
@@ -0,0 +1,144 @@
+ * but WITHOUT ANY WARRANTY; without even the implied warranty of
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+ * GNU General Public License for more details.
+ *
+ * You should have received a copy of the GNU General Public License
+ * along with this program.  If not, see <http://www.gnu.org/licenses/>.
+ */
+
+import React, { useState } from 'react';
+import { Route, Switch, useHistory } from 'react-router';
+import { useSelection } from '../../utils/hooks/useSelection';
+import { useGlobalAppState } from '../GlobalApp';
+import { useReference } from '../../utils/hooks/useReference';
+import { useActiveSiteId } from '../../utils/hooks/useActiveSiteId';
+import { usePreviewState } from '../../utils/hooks/usePreviewState';
+import { useEnv } from '../../utils/hooks/useEnv';
+import { getSystemLink } from '../LauncherSection';
+import SiteToolsApp, { Tool } from './SiteToolsApp';
+import Widget from '../Widget';
+import Box from '@material-ui/core/Box';
+import LauncherOpenerButton from '../LauncherOpenerButton';
+import EmptyState from '../SystemStatus/EmptyState';
+import { FormattedMessage } from 'react-intl';
+import { urlDrivenStyles } from './styles';
+import Suspencified from '../SystemStatus/Suspencified';
+
+interface UrlDrivenSiteToolsProps {
+  footerHtml: string;
+}
+
+export default function UrlDrivenSiteTools(props: UrlDrivenSiteToolsProps) {
+  const { footerHtml } = props;
+  const [width, setWidth] = useState(240);
+  const history = useHistory();
+  const [activeToolId, setActiveToolId] = useState(history.location.pathname);
+  const baseUrl = useSelection<string>((state) => state.env.authoringBase);
+  const [{ openSidebar }] = useGlobalAppState();
+  const tools: Tool[] = useReference('craftercms.siteTools')?.tools;
+  const site = useActiveSiteId();
+  const { previewChoice } = usePreviewState();
+  const { authoringBase } = useEnv();
+  const classes = urlDrivenStyles();
+
+  history.listen((location) => {
+    setActiveToolId(location.pathname);
+  });
+
+  const onNavItemClick = (id: string) => {
+    history.push(id);
+  };
+
+  const onBackClick = () => {
+    window.location.href = getSystemLink({
+      site,
+      previewChoice,
+      authoringBase,
+      systemLinkId: 'preview'
+    });
+  };
+
+  return (
+    <SiteToolsApp
+      site={site}
+      width={width}
+      onWidthChange={setWidth}
+      onNavItemClick={onNavItemClick}
+      onBackClick={onBackClick}
+      activeToolId={activeToolId}
+      footerHtml={footerHtml}
+      openSidebar={openSidebar}
+      tools={tools}
+    >
+      <Switch>
+        {tools?.map((tool) => (
+          <Route
+            key={tool.url}
+            path={`/${tool.url}`}
+            render={() => {
+              return (
+                <Suspencified>
+                  <Widget {...tool.widget} extraProps={{ embedded: false }} />
+                </Suspencified>
+              );
+            }}
+          />
+        ))}
```

Comment:
Looks like this is duplicate on the site tools app. We need a way to factor/abstract tool rendering to a single place.